crush the vectorized method resistance

Sacha0 opened the pull request:
Fly the red flag and grant them no quarter.

Fixes #22959. Needs news? Best!

Sacha0 committed:
Deprecate remaining zero-preserving vectorized methods over SparseVectors.

Sacha0 committed:
Deprecate remaining not-zero-preserving vectorized methods over SparseVectors.

tkelman commented on base/sparse/sparsevector.jl:
```diff
@@ -1094,16 +1083,6 @@ macro unarymap_z2nz(op, TF)
-           :acos, :acot, :acosd, :acotd,
-           :cosh, :csch, :coth, :sech,
-           :acsch, :asech]
-    @eval @unarymap_z2nz $(op) Number
```
we could also deprecate this macro? (or remove if not used outside of base)

Sacha0 replied:
Absolutely, chances are most of the `Xmap_YZ` macros/functions for `SparseVector`s can be cleaned out with some care :). Would a separate pull request be alright?

martinholters commented:
https://www.youtube.com/watch?v=8CP9dg38cAI

fredrikekre commented:
Perhaps add a NEWS bullet and this should be good to go?

Sacha0 committed:
Add NEWS entry for deprecation of remaining vectorized methods and update news PR links.

Sacha0 commented:
> Perhaps add a NEWS bullet and this should be good to go?

Thanks for the bump! Added a commit with news for this pull request and updated news PR links. Best!

fredrikekre reviewed (approved):
Travis is one timeout and Arnoldi.

Sacha0 commented:
Thanks all! :)

JeffBezanson commented:
This is truly a glorious moment.

StefanKarpinski commented:
Wow, is this really the last of the vectorized methods? That is momentous.

Sacha0 commented:
I suspect pockets of resistance remain, hopefully to be smoked out prior to The Great Feature Freeze.

StefanKarpinski commented:
Hmm. I wonder if there's a way to automatically test for vectorized methods...

StefanKarpinski commented:
Ok, I found some lingerers by the very high tech means of grepping for `vectoriz` in the code base:

https://github.com/JuliaLang/julia/blob/0f4229d2e744375255fa1b27cb180f678ebf3e76/base/dates/io.jl#L561-L580
